src/InputWithLabel.jsx

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from "react";
+
+const InputWithLabel = ({id, type, name, value, isFocused, onChange, children}) => {
+    const inputRef = useRef();
+    useEffect(()=> {
+        if (isFocused && inputRef.current){
+            inputRef.current.focus();
+        }
+    }, [isFocused])
```

Comment:
Remove the dependency list altogether so this fires on every render. Otherwise, the `useEffect` is never being called since the value of `isFocused` never changes.